internal/host/validator.go

```diff
@@ -94,18 +95,32 @@ func (c *validationContext) loadInfraEnv() error {
 	return err
 }
 
-func (c *validationContext) loadInventory() error {
-	if c.host.Inventory != "" {
-		var inventory models.Inventory
-		err := json.Unmarshal([]byte(c.host.Inventory), &inventory)
+func (c *validationContext) getInventory(host *models.Host) (inventory *models.Inventory, err error) {
```

Comment:
I feel like this would be better as a `Get` function on the InventoryCache type.

It doesn't look like this uses anything from the validation context other than the cache.

Reply:
OK